File: JavaScript/String/L_findingCommonChar.js
var count = function (word) {
    const frequency = Array(26).fill(0);
    for (let char of word) {
        frequency[char.charCodeAt(0) - "a".charCodeAt(0)]++;
    }
    return frequency;
};

var intersection = function (freq1, freq2) {
    return freq1.map((f1, idx) => Math.min(f1, freq2[idx]));
};

var commonChars = function (words) {
    let last = count(words[0]);
    for (let i = 1; i < words.length; i++) {
        last = intersection(last, count(words[i]));
    }

    const result = [];
    for (let i = 0; i < 26; i++) {
        while (last[i] > 0) {
            result.push(String.fromCharCode(i + "a".charCodeAt(0)));
            last[i]--;
        }
    }

    return result;
};

File: Python/Strings/L_findingCommonChar.py
def count(word):
    frequency = [0] * 26
    for char in word:
        frequency[ord(char) - ord('a')] += 1
    return frequency

def intersection(freq1, freq2):
    return [min(f1, f2) for f1, f2 in zip(freq1, freq2)]

def commonChars(words):
    # Initialize the frequency array with the count of the first word
    last = count(words[0])
    
    # Update the frequency array by taking intersections with counts of remaining words
    for word in words[1:]:
        last = intersection(last, count(word))
    
    # Construct the result list
    result = []
    for i in range(26):
        result.extend([chr(i + ord('a'))] * last[i])
    
    return result 